### ISSUE TITLE
No support for Plugins

### ISSUE THREAD

**Issue by jfreak53** [JoomShaper/SP-Page-Builder#38]
Page Builder has zero support for plugins inside it's content. This is seriously limiting and has forced me to edit your code just to make it work.

It's so simple to make your addons support plugin code, I don't understand why you don't make it work?

http://cdn.microthosting.com/imgs/Screenshot_at_2016-07-28_16-29-59.png

I had to edit your tabs just to make it support plugins so I could include a local video player in a tab. Not even text content supports plugins. A simple one line of code would fix this, I feel this is a serious short fall.

**Comment by nayemmajhar** [comment]
Do you know how override Page Builder addons? Please check here https://www.joomshaper.com/documentation/joomla-extensions/sp-page-builder#documentation-17 Hope it will solve your problem.

note: In upcoming version, we will add plugins support for developer.

**Comment by nayemmajhar** [comment]
@jfreak53 I will check this trick later. Thanks

**Comment by pepperstreet** [comment]
+1 

---

I have found my older support forum topic in PB v1 category. Now, I have moved it to the PB v2 category and cross-linked with this Github issue. 
https://www.joomshaper.com/forums/text-html-addons-prepare-content-run-plugins

**Comment by pepperstreet** [comment]
Any news on this? Maybe in upcoming PageBuilder v3 ?

**Comment by ideaalab** [comment]
Looks like an easy fix, as @jfreak53 mentioned.
Why dont you add it for the next version?
We use many different plugins, like forms, and insert content that not work.

**Comment by pepperstreet** [comment]
As time rushes by... PB v**3.1.3** is there, but no support nor an option for "content plugins" :(

**Comment by jfreak53** [comment]
Agreed! It's frustrating because it's SO simple to add for them! I have to re-add it to my sites every time I upgrade pagebuilder, I'm a paid developer of the Pro version, this is such a simple feature it should already be there!!

So simple, simply add this to each text output line for each addon:

`$output .= JHtml::_('content.prepare', $text);`

Why is that so hard to do to all the addons?

**Comment by pepperstreet** [comment]
@jfreak53 
> Why is that so hard to do to all the add ons? 

Probably it is not the best way to implement it "everywhere". At least not as a permanent default. I could imagine that it might produce too many calls and code execution. Possibly performance or memory issues in the worst case... (just thinking loud, I just know enough to be dangerous ;) ) 

I would like to see an optional feature, which can be turned ON/OFF. Like we have in J! core CustomHTML module parameters. 
I am not sure which structural element or location is the best for PageBuilder. Probably per AddOn. 

Maybe the @JoomShaper devs can shed some light on the technical side?

**Comment by pepperstreet** [comment]
FYI – the PB **Joomla article integration** feature is some sort of a workaround. 
Joomla core articles designed with PB editor seem to prepare all content plugins ;)

**Comment by natiki** [comment]
I have to agree with the rest of the posters here. The fact that you cannot load content via plugins makes SP-PageBuilder pretty much useless. I now have a site that was built previously using PB and trying to do something as simple as embed a RSForm using:

https://www.rsjoomla.com/support/documentation/rsform-pro/plugins-and-modules/content-plugin-plgcontent-display-the-form-in-an-article.html

does not work and all I get is the shortcode `{rsform 3}` For whatever reason the Joomla Article Integration work around is not working either for me.

So if anyone has successfully embedded a RSForm I would love to hear from you. I have tried embedding a Joomla Module using their module plugin with the same results.

**Comment by MarkizMdS** [comment]
> Agreed! It's frustrating because it's SO simple to add for them! I have to re-add it to my sites every time I upgrade pagebuilder, I'm a paid developer of the Pro version, this is such a simple feature it should already be there!!
> 
> So simple, simply add this to each text output line for each addon:
> 
> `$output .= JHtml::_('content.prepare', $text);`
> 
> Why is that so hard to do to all the addons?


Could you clarify in more detail exactly where to insert this code? I tried as you wrote, but it didn’t work out for me.

**Comment by jfreak53** [comment]
It depends entirely on the addon you wish to add it to. You simply replace the output with the `content.prepare` part of JTML. Which addon do you wish to add this to?
